main/tests/server/src/com/google/refine/model/recon/StandardReconConfigTests.java

```diff
@@ -564,11 +566,94 @@ public void BatchReconTestError() throws Exception {
+                    "             \"propertyName\": \"Director\",\n" +
+                    "             \"propertyID\": \"P57\"\n" +
+                    "           }\n" +
+                    "        ]}";
+            StandardReconConfig config = StandardReconConfig.reconstruct(configJson);
+            StandardReconConfig.StandardReconJob job = new StandardReconConfig.StandardReconJob();
+            job.text = "david lynch";
+            job.code = "{\"query\":\"david lynch\",\"type\":\"Q11424\",\"properties\":[{\"pid\":\"P57\",\"v\":\"david lynch\"}],\"type_strict\":\"should\"}";
+            List<ReconJob> jobList = new ArrayList<ReconJob>();
+            jobList.add(job);
+
+            // calling the batchRecon
+            List<Recon> returnReconList = config.batchRecon(jobList, 1000000000);
+
+            RecordedRequest request1 = server.takeRequest();
+            assertNotNull(request1);
+            String query = request1.getBody().readUtf8Line();
+
+            // assertions
+            assertNotNull(returnReconList.get(0).error);
+            assertEquals(returnReconList.get(0).error, "Read timed out");
+            assertNotNull(returnReconList);
+        }
+    }
+
+    @Test
+    public void batchReconTestDNSError() throws Exception {
+        HttpUrl url = HttpUrl.parse("https://hewsjsajsajk.com/search?q=ujdjsaoiksa");
+
+        String configJson = " {\n" +
+                "        \"mode\": \"standard-service\",\n" +
+                "        \"service\": \"" + url + "\",\n" +
+                "        \"identifierSpace\": \"http://www.wikidata.org/entity/\",\n" +
+                "        \"schemaSpace\": \"http://www.wikidata.org/prop/direct/\",\n" +
+                "        \"type\": {\n" +
+                "                \"id\": \"Q11424\",\n" +
+                "                \"name\": \"film\"\n" +
+                "        },\n" +
+                "        \"autoMatch\": true,\n" +
+                "        \"columnDetails\": [\n" +
+                "           {\n" +
+                "             \"column\": \"director\",\n" +
+                "             \"propertyName\": \"Director\",\n" +
+                "             \"propertyID\": \"P57\"\n" +
+                "           }\n" +
+                "        ]}";
+        StandardReconConfig config = StandardReconConfig.reconstruct(configJson);
+        StandardReconConfig.StandardReconJob job = new StandardReconConfig.StandardReconJob();
+        job.text = "david lynch";
+        job.code = "{\"query\":\"david lynch\",\"type\":\"Q11424\",\"properties\":[{\"pid\":\"P57\",\"v\":\"david lynch\"}],\"type_strict\":\"should\"}";
+        List<ReconJob> jobList = new ArrayList<ReconJob>();
+        jobList.add(job);
+
+        List<Recon> returnReconList = config.batchRecon(jobList, 1000000000);
+        // assertions
+        // assertEquals(returnReconList.get(0).error, url.host() + ": nodename nor servname provided, or not known");
```

Comment:
Here we can have a comment explaining that the error message is unstable and system-dependent, so we are not asserting for its exact contents.